### ISSUE TITLE
missing indent in def train(...) in `imagenet`

### ISSUE THREAD

**Issue by HearyShen** [pytorch/examples#638]
https://github.com/pytorch/examples/blob/ee964a2eeb41e1712fe719b83645c79bcbd0ba1a/imagenet/main.py#L284

It seems a missing indent in imagenet train(...) function.

`/example/imagenet/main.py`, line 282 to 284.

```python
        if args.gpu is not None:
            images = images.cuda(args.gpu, non_blocking=True)
        target = target.cuda(args.gpu, non_blocking=True)
```
The default value of `args.gpu` is None.
When `args.gpu` is not specified (default as None), `images` tensor is not moved to cuda, which is reasonable. But, why `target` tensor is still moved to cuda? Is there a missing tab indent?

In this example, the `model` is always moved to cuda, so the `outputs` is in cuda. Always moving `target` tensor to cuda can avoid causing error for the following `loss = criterion(outputs, targets)`. If this is the consideration, then why `images` tensor is kept in cpu?

**Comment by soumith** [comment]
the reason it is not moved to `.cuda(args.gpu)` is because we expect the `DataParallel` or `DistributedDataParallel` to move it to relevant GPUs more uniformly. So, instead of moving `images` to GPU0, it will be more like `images[0:16]` are moved to GPU0, `images[17:32]` are moved to GPU1, etc.
